JumpscaleLibs/sal/reservation_chatflow/reservation_chatflow.py

```diff
@@ -1032,10 +1033,22 @@ def gateway_list(self, bot, currency=None):
                 areaname = getattr(g.location, area)
                 if areaname not in unknowns:
                     location.append(areaname)
+            currencies = list()
+
+            farm_id = g.farm_id
+            if farm_id not in farms:
+                farms[farm_id] = j.sal.zosv2._explorer.farms.get(farm_id)
+
+            addresses = farms[farm_id].wallet_addresses
+            for address in addresses:
+                if address.asset not in currencies:
+                    currencies.append(address.asset)
+
             if g.free_to_use:
-                reservation_currency = "FreeTFT"
-            else:
-                reservation_currency = "TFT"
+                if "FreeTFT" not in currencies:
+                    currencies.append("FreeTFT")
+            reservation_currency = currencies
```

Comment:
shouldnt this be something like 
```suggestion
            reservation_currency = ", ".join(currencies)
```